main.tf

```diff
@@ -0,0 +1,153 @@
+}
+
+// This enables app engine, but more importantly also enables Datastore
+resource "google_app_engine_application" "app" {
+  project     = "${google_project.project.project_id}"
+  location_id = "${var.region}"
+}
+
+resource "google_project_service" "compute" {
+  project = "${google_project.project.project_id}"
+  service = "compute.googleapis.com"
+
+  disable_dependent_services = true
+  disable_on_destroy         = false
+}
+
+resource "google_project_service" "container" {
+  project = "${google_project.project.project_id}"
+  service = "container.googleapis.com"
+
+  disable_dependent_services = true
+}
+
+resource "google_project_service" "logging" {
+  project = "${google_project.project.project_id}"
+  service = "logging.googleapis.com"
+
+  disable_dependent_services = true
+  disable_on_destroy         = false
+}
+
+resource "google_compute_network" "k8s" {
+  depends_on              = ["google_project_service.compute"]
+  name                    = "k8s"
+  auto_create_subnetworks = "true"
+  project                 = "${google_project.project.project_id}"
+}
+
+// GKE
+
+# service account for compute engine instances
+resource "google_service_account" "compute" {
+  account_id   = "compute"
+  display_name = "Compute Engine service account"
+  project      = "${google_project.project.project_id}"
+}
+
+resource "google_project_iam_member" "compute" {
+  project = "${google_project.project.project_id}"
+  role    = "roles/editor"
+  member  = "serviceAccount:${google_service_account.compute.email}"
+}
+
+resource "google_container_cluster" "runner-benchmark" {
+  name                     = "runner-benchmark"
+  description              = "Kubernetes Cluster - Dev Benchmark environment"
+  location                 = "${var.region}"
+  min_master_version       = "${var.k8s_min_master_version}"
+  initial_node_count       = 1
+  remove_default_node_pool = true
+  project                  = "${google_project.project.project_id}"
+  network                  = "${google_compute_network.k8s.self_link}"
+
+  // Basic auth is disabled by setting user/pass to empty strings
+  master_auth {
+    username = ""
+    password = ""
+
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+
+  addons_config {
+    kubernetes_dashboard {
+      disabled = true
+    }
+  }
+}
+
+resource "google_container_node_pool" "main-node-pool" {
+  name       = "main-node-pool"
+  location   = "${var.region}"
+  cluster    = "${google_container_cluster.runner-benchmark.name}"
+  node_count = 1
+  project    = "${google_project.project.project_id}"
+  version    = "${var.k8s_min_master_version}"
+
+  lifecycle {
+    ignore_changes = ["node_count", "version"]
+  }
+
+  autoscaling {
+    min_node_count = 1
+    max_node_count = 5
+  }
+
+  management {
+    auto_repair  = true
+    auto_upgrade = true
+  }
+
+  node_config {
+    machine_type = "n1-standard-1"
+
+    oauth_scopes = [
+      "compute-rw",
+    ]
```

Comment:
The node pool gets destroyed and re-created on each create as some scopes are missing here. Adding `logging-write` and `monitoring` fixes this.